Add taxpasta_merge to taxprofiler (#241)
nf-core/taxprofiler: docs/output.md

- [MALT](#malt) - Sequence alignment and analysis tool designed for processing high-throughput sequencing data, especially in the context of metagenomics
- [MetaPhlAn3](#metaphlan3) - Genome-level marker gene based taxonomic classifier
- [mOTUs](#motus) - Tool for marker gene-based OTU (mOTU) profiling.

> - [TAXPASTA](#taxpasta) - Tool to standardise taxonomic profiles as well as merge profiles across samples for the same classifier/profiler.

Reviewer: ```suggestion
- [TAXPASTA](#taxpasta) - Tool to standardise taxonomic profiles as well as merge profiles across samples from the same database and classifier/profiler.
```

The resulting HTML files can be loaded into your web browser for exploration. Each file will have a dropdown to allow you to switch between each sample aligned against the given database of the tool.

### TAXPASTA

> [TAXPASTA](https://github.com/taxprofiler/taxpasta) is a python package added as a module in nf-core/taxprofiler that can standardise and merge two or more taxonomic profiles across samples into one single table..

Reviewer: ```suggestion
[TAXPASTA](https://github.com/taxprofiler/taxpasta) that standardises and merge two or more taxonomic profiles across samples into one single table. It supports multiple different classifiers simplifying taxonomic classification results between tools and databases.
```

<details markdown="1">
<summary>Output files</summary>

- `taxpasta`

>   - `<db_name>.<sample_id>*.{tsv,csv,arrow,parquet,biom}`: A list of taxonomic profiler output files. The standard format is the `tsv`. The first column describes the taxonomy ID and the rest of the columns describe the read counts for each sample.

Reviewer: It shouldn't produced a sample id,
```suggestion
  - `<tool>_<database>*.{tsv,csv,arrow,parquet,biom}`: Standardised taxon table containing multiple samples. The standard format is the `tsv`. The first column describes the taxonomy ID and the rest of the columns describe the read counts for each sample.
```

Author: You are right, thanks!

  </details>

> The following report files are used for the taxpasta step:

Reviewer: ```suggestion
These are files will likely be the most useful files for the comparison of differences in classification between different tools or building consensuses, with the caveat they have slightly less information than the actual output from each tool (which may have non-standard information e.g. taxonomic rank, percentage of hits, abundance estimations).

The following report files are used as input for the taxpasta step:
```

Reviewer: Should also add the sentence:

```
Please aware the outputs of each tool's standardised profile _may not_ be directly comparable. Some may report raw read counts, or others may report abundance information. Please see the list tomorrow of which information is used for each tool. 
```

> - Bracken: `<sample>_<db_name>.tsv`

Reviewer: Taxpasta displays the `new_est_reads` column? @sofstam @Midnighter can you confirm?

We need make it clear _which_ value (count? percent? abundance value) is being displayed for each tool (please check my comments below for all profilers).

Reviewer: Should probably be documented in taxpsata too tbh...

Reviewer 2: It's the `new_est_reads`, yes. And I agree, that should be documented on taxpasta.

> - Centrifuge: `<sample_id>.centrifuge.txt`

Reviewer: ```suggestion
- Centrifuge: `<sample_id>.centrifuge.txt`: taxpasta uses the 'direct_assigned_reads' column for the standardised profile.
```

> - Diamond: `<sample_id>`

Reviewer: ```suggestion
- Diamond: `<sample_id>.tsv`: `: taxpasta summarises number of reads per NCBI taxonom ID standardised profile.
```

> - Kaiju: `<sample_id>_<db_name>.kaijutable.txt`

Reviewer: ```suggestion
- Kaiju: `<sample_id>_<db_name>.kaijutable.txt`: taxpasta uses the 'reads' column from kaiju2table standardised profile.
```

> - KrakenUniq: `<sample_id>_<db_name>.report.txt`

Reviewer: ```suggestion
- KrakenUniq: `<sample_id>_<db_name>.report.txt``: taxpasta uses the 'reads' column for the standardised profile.
```

> - Kraken2: `<sample_id>_<db_name>.report.txt`

Reviewer: ```suggestion
- Kraken2: `<sample_id>_<db_name>.report.txt`: taxpasta uses the 'direct_assigned_reads' column for the standardised profile.
```

> - MALT: `<sample_id>.txt.gz`

Reviewer: ```suggestion
- MALT: `<sample_id>.txt.gz`: taxpasta uses the 'count' (second) column from the output of MEGAN6's rma2info for the standardised profile.
```

> - MetaPhlAn3: `<sample_id>_profile.txt`

Reviewer: ```suggestion
- MetaPhlAn3: `<sample_id>_profile.txt`: taxpasta uses the 'relative_abundance' column for the standardised profile.
```

Reviewer 2: Multiplied with a fixed number to yield an integer.

Reviewer: @sofstam please update to this effect :)

> - mOTUs: `<sample_id>.out`

Reviewer: ```suggestion
- mOTUs: `<sample_id>.out`: taxpasta uses the 'read_count' column for the standardised profile.
```

The following report files are used for the taxpasta step:

- Bracken: `<sample>_<db_name>.tsv` Taxpasta used the `new_est_reads` column for the standardised profile.
- Centrifuge: `<sample_id>.centrifuge.txt` Taxpasta uses the `direct_assigned_reads` column for the standardised profile.
- Diamond: `<sample_id>` Taxpasta summarises number of reads per NCBI taxonomy ID standardised profile.
- Kaiju: `<sample_id>_<db_name>.kaijutable.txt` Taxpasta uses the `reads` column from kaiju2table standardised profile.
- KrakenUniq: `<sample_id>_<db_name>.report.txt` Taxpasta uses the `reads` column for the standardised profile.
- Kraken2: `<sample_id>_<db_name>.report.txt` Taxpasta uses the `direct_assigned_reads` column for the standardised profile.
- MALT: `<sample_id>.txt.gz` Taxpasta uses the `count` (second) column from the output of MEGAN6's rma2info for the standardised profile.
- MetaPhlAn3: `<sample_id>_profile.txt` Taxpasta uses the `relative_abundance` column multiplied with a fixed number to yield an integer for the standardised profile.
- mOTUs: `<sample_id>.out` Taxpasta uses the `read_count` column for the standardised profile.

> 

Reviewer: ```suggestion
Please aware the outputs of each tool's standardised profile _may not_ be directly comparable between each tool. Some may report raw read counts, whereas others may report abundance information. Please always refer to the list above, for which information is used for each tool. 

```

The resulting HTML files can be loaded into your web browser for exploration. Each file will have a dropdown to allow you to switch between each sample aligned against the given database of the tool.

### TAXPASTA

> [TAXPASTA](https://github.com/taxprofiler/taxpasta) that standardises and merge two or more taxonomic profiles across samples into one single table. It supports multiple different classifiers simplifying taxonomic classification results between tools and databases.

Reviewer: ```suggestion
[TAXPASTA](https://github.com/taxprofiler/taxpasta) standardises and merges two or more taxonomic profiles across samples into one single table. It supports multiple different classifiers simplifying comparison of taxonomic classification results between tools and databases.
```

The following report files are used for the taxpasta step:

- Bracken: `<sample>_<db_name>.tsv` Taxpasta used the `new_est_reads` column for the standardised profile.
- Centrifuge: `<sample_id>.centrifuge.txt` Taxpasta uses the `direct_assigned_reads` column for the standardised profile.
- Diamond: `<sample_id>` Taxpasta summarises number of reads per NCBI taxonomy ID standardised profile.
- Kaiju: `<sample_id>_<db_name>.kaijutable.txt` Taxpasta uses the `reads` column from kaiju2table standardised profile.
- KrakenUniq: `<sample_id>_<db_name>.report.txt` Taxpasta uses the `reads` column for the standardised profile.
- Kraken2: `<sample_id>_<db_name>.report.txt` Taxpasta uses the `direct_assigned_reads` column for the standardised profile.
- MALT: `<sample_id>.txt.gz` Taxpasta uses the `count` (second) column from the output of MEGAN6's rma2info for the standardised profile.
- MetaPhlAn3: `<sample_id>_profile.txt` Taxpasta uses the `relative_abundance` column multiplied with a fixed number to yield an integer for the standardised profile.
- mOTUs: `<sample_id>.out` Taxpasta uses the `read_count` column for the standardised profile.

> Please aware the outputs of each tool's standardised profile _may not_ be directly comparable between each tool. Some may report raw read counts, whereas others may report abundance information. Please always refer to the list above, for which information is used for each tool.

Reviewer: ```suggestion
> ⚠️ Please aware the outputs of each tool's standardised profile _may not_ be directly comparable between each tool. Some may report raw read counts, whereas others may report abundance information. Please always refer to the list above, for which information is used for each tool.
```
